indico/modules/events/registration/controllers/management/reglists.py

```diff
@@ -764,6 +775,70 @@ def _process(self):
         return jsonify_form(form, disabled_until_change=False, submit=_('Reject'), message=message)
 
 
+class RHRegistrationsBasePrice(RHRegistrationsActionBase):
+    """Edit the base price of the selected registrations."""
+
+    def _process(self):
+        form = RegistrationBasePriceForm(base_price=self.regform.base_price, currency=self.regform.currency,
+                                         registration_id=[reg.id for reg in self.registrations])
+        if form.validate_on_submit():
+            log_fields = {
+                'base_price': {'title': 'Registration Fee', 'type': 'string'},
+                'state': 'State'
+            }
+            num_skipped = 0
+            num_updates = 0
+            for reg in self.registrations:
+                prev_state = reg.state
+                if form.apply_complete.data and reg.state == RegistrationState.complete and not reg.base_price:
+                    reg.state = RegistrationState.unpaid
+                elif reg.state != RegistrationState.unpaid:
+                    num_skipped += 1
+                    continue
+                new_price = {
+                    'remove': 0,
+                    'default': self.regform.base_price,
+                    'custom': form.base_price.data
+                }[form.action.data]
+                changes = {}
+                if reg.base_price != new_price or reg.currency != self.regform.currency:
+                    changes['base_price'] = (format_currency(reg.base_price, reg.currency, locale='en_GB'),
+                                             format_currency(new_price, self.regform.currency, locale='en_GB'))
+                    reg.base_price = new_price
+                    reg.currency = self.regform.currency
+                if not reg.price:
+                    reg.state = RegistrationState.complete
+                if prev_state != reg.state:
+                    changes['state'] = (prev_state, reg.state)
+                    notify_registration_state_update(reg, from_management=True)
```

Comment:
We should also trigger the registration state change signal. Note that this already includes logging (in `log_registration_updated`) which is a bit misleading for an unpaid<->complete transitions not involving actual payments) so you probably need to add a `no_log` or similar kwarg in the signal to prevent logging in this particular case.